ssl/statem/statem_clnt.c

```diff
@@ -2240,11 +2240,9 @@ MSG_PROCESS_RETURN tls_process_certificate_request(SSL *s, PACKET *pkt)
             goto err;
         }
 
-        /* Clear certificate digests and validity flags */
-        for (i = 0; i < SSL_PKEY_NUM; i++) {
-            s->s3->tmp.md[i] = NULL;
+        /* Clear certificate validity flags */
+        for (i = 0; i < SSL_PKEY_NUM; i++)
             s->s3->tmp.valid_flags[i] = 0;
```

Comment:
Every time I see this kind of thing, I weep. :)
```
{ declare uint32_t *fp; }
for (fp = s->s3->tmp.valid_flags, i = SSL_PKEY_NUM; --i >= 0; fp++)
    *fp = 0;
```

Back when things like this counted ... :)